enhancements/ingress/set-replace-append-delete-deny-headers.md

```diff
@@ -0,0 +1,956 @@
+    // +listType=map
+    // +listMapKey=name
+    // +optional
+    // +kubebuilder:validation:MaxItems=64
+    // +kubebuilder:validation:XValidation:rule=`self.all(key, key.action.type == "Delete" || (has(key.action.set) && key.action.set.value.matches('^(?:%(?:%|(?:\\{[-+]?[QXE](,[-+]?[QXE])*\\})?\\[(?:req\\.hdr\\([0-9A-Za-z-]+\\)|ssl_c_der)(?:,(?:lower|base64))*\\])|[^%[:cntrl:]]*)$')))`,
+message="Either the header value provided is not in correct format or the sample fetcher/converter specified is not allowed. The dynamic header value will be interpreted as an HAProxy
+format string as defined in http://cbonte.github.io/haproxy-dconv/2.6/configuration.html#8.2.6 and may use HAProxy's %[] syntax and otherwise must be a valid HTTP header value as
+defined in https://datatracker.ietf.org/doc/html/rfc7230#section-3.2 Sample fetchers allowed are req.hdr, ssl_c_der. Converters allowed are lower, base64."
+    Request []IngressControllerHTTPHeader `json:"request,omitempty"`
+}
+```
+
+```go
+// IngressControllerHTTPHeader specifies configuration for setting or deleting an HTTP header.
+type IngressControllerHTTPHeader struct {
+    // name specifies the name of a header to set or delete.  Its value must be a valid HTTP header.
+    // name as defined in RFC 2616 section 4.2.
+    // +kubebuilder:validation:Required
+    // +kubebuilder:validation:MinLength=1
+    // +kubebuilder:validation:MaxLength=256
+    // +kubebuilder:validation:Pattern="^[-!#$%&'*+.0-9A-Z^_`a-z|~]+$"
+    // +kubebuilder:validation:XValidation:rule="self.lowerAscii() != 'strict-transport-security'",message="strict-transport-security may not be set via header actions"
+    // +kubebuilder:validation:XValidation:rule="self.lowerAscii() != 'proxy'",message="proxy may not be set/delete via header actions"
+    // +kubebuilder:validation:XValidation:rule="self.lowerAscii() != 'host'",message="host may not be set/delete via header actions"
+    Name string `json:"name"`
+
+    // action specifies actions to perform on headers, such as setting or deleting headers.
+    // +kubebuilder:validation:Required
+    Action IngressControllerHTTPHeaderActionUnion `json:"action"`
+}
+```
+
+```go
+// IngressControllerHTTPHeaderActionUnion specifies an action to take on an HTTP header.
+// +union
+type IngressControllerHTTPHeaderActionUnion struct {
+    // type defines the type of the action to be applied on the header.
+    // Possible values are Set and Delete.
+    // Set allows you to set http request and response headers.
+    // Delete allows you to delete http request and response headers
+    // +unionDiscriminator
+    // +kubebuilder:validation:Enum:=Set;Delete
+    // +kubebuilder:validation:Required
+    Type IngressControllerHTTPHeaderActionType `json:"type"`
+
+    // set specifies how the HTTP header should be set.
+    // If this field is empty, no action is taken.
+    // +optional
+    // +unionMember
+    Set *IngressControllerSetHTTPHeader `json:"set,omitempty"`
```

Comment:
This is clunky.  I don't know why IngressControllerSetHTTPHeader.value was not just put here and named Value.  If it's too late to change, then it's fine.